### DISCUSSION TITLE
Refresh_token login remove in supabasejs 2.0+?

### DISCUSSION THREAD

**Discussion by KULTI1995** [supabase/supabase#9495] [Questions]
login using refresh_token was removed, along with the introduction of supabase 2.0+?
`await supabase.auth.sign({refreshToken});`

**Comment by Sukikiroi** [comment]
Hello, 
no it still there 
Supabase setup a timer to refresh the token just before it expires.

And by default, the

autoRefreshToken: true,

**Reply by Sukikiroi** [reply]
also can you tel me pls is this for Dart 
cause it seems there's Bug on Dart repo related to This

**Reply by KULTI1995** [reply]
Through what function? Because signIn no longer exists.

**Reply by Jaarson** [reply]
Bumping up, I'm using refresh token in a URL to open an Electron app.
What function should I use to sign in when I only have the refresh token?

**Comment by GaryAustin1** [comment]
I think refreshSession() with a refresh token is what you want.
![image](https://user-images.githubusercontent.com/54564956/213025795-b20d4730-370d-48ad-89cc-ef7230a1ff44.png)
But not tested.

**Reply by j4w8n** [reply]
I never thought about `refreshSession()` being used to login, but it def will as long as you pass in a valid refresh token.

Keep in mind it will trigger a `TOKEN_REFRESHED` event, not `SIGNED_IN`.
